### DISCUSSION TITLE
Secret scanning gives false positive for Google Firebase API key

### DISCUSSION THREAD

**Discussion by boocockp** [community/community#58734] [Code Security]
### Select Topic Area

Product Feedback

### Body

Web apps that use Google Firebase need to embed a unique API key in the client code just to identify which app is being used.  This is flagged by secret scanning, but it has to be sent in clear to every user of the app, and it is not used for authentication or to control access.  Google documentation says it is not intended to be secret.  
More info at [https://firebase.google.com/docs/projects/learn-more#config-files-objects](https://firebase.google.com/docs/projects/learn-more#config-files-objects)

**Comment by victorpreston** [comment]
Yes, you are correct. When using Google Firebase for web apps, it is necessary to include a unique API key in the client code to identify the app. This API key is not intended to be kept secret and is not used for authentication or access control purposes.
While secret scanning tools may flag the presence of the API key in the code, it is a necessary step in the configuration of Firebase for web apps. The API key is used to establish a connection between the app and the Firebase services.

**Comment by setoelkahfi** [comment]
Can someone from GitHub chime in? We're currently considering whether to put effort into hiding the Firebase API key from our `.plist` file.

**Reply by junaidahmed12095** [reply]
Okay

**Reply by SPodjasek** [reply]
As nobody from GitHub has picked this up, I guess I'll have to close those issues as false positives giving reference to this page: https://firebase.google.com/support/guides/security-checklist#api-keys-not-secret
